### ISSUE TITLE
[Buttons] Buttons.md 中介绍的使用方式错误

### ISSUE THREAD

**Issue by jimie-chaos** [material-components/material-components-android#3526]
As mentioned in the file.
I would like to modify all Button styles to have a green background and black font
wrong:
```xml
<style name="Theme.App" parent="Theme.Material3.*">
...
<style name="ThemeOverlay.App.Button" parent="">
    <!-- green -->
    <item name="colorPrimary">@color/teal_700</item>
    <!-- black -->
    <item name="colorOnPrimary">@color/black</item>
</style>
```
right:
```xml
<style name="ThemeOverlay.App.Button" parent="">
    <!-- green -->
    <item name="colorContainer">@color/teal_700</item>
    <!-- black -->
    <item name="colorOnContainer">@color/black</item>
</style>
```
and:
My `colorPrimary` cannot be used correctly,maybe.
my theme:
```
<style name="Theme.testApllication" parent="Theme.Material3.Light.NoActionBar">
    <!-- Primary brand color. -->
    <!-- purple -->
    <item name="colorPrimary">@color/purple_500</item>
    <item name="colorPrimaryVariant">@color/purple_700</item>
    <!-- white -->
    <item name="colorOnPrimary">@color/white</item>
    <!-- Secondary brand color. -->
    <item name="colorSecondary">@color/teal_200</item>
    <item name="colorSecondaryVariant">@color/teal_700</item>
    <item name="colorOnSecondary">@color/black</item>
    <!-- Status bar color. -->
    <item name="android:statusBarColor">?attr/colorPrimaryVariant</item>
    <item name="toolbarStyle">@style/Widget.App.Toolbar</item>
    <item name="borderlessButtonStyle">@style/Widget.App.Button.TextButton</item>
    <item name="materialButtonOutlinedStyle">@style/Widget.App.Button.OutlinedButton</item>
    <item name="materialButtonStyle">@style/Widget.App.Button</item>
    <!-- Customize your theme here. -->
</style>
```

only use **Theme.testApllication**
 and no **borderlessButtonStyle**,**materialButtonOutlinedStyle**,**materialButtonStyle**:
<img width="257" alt="image" src="https://github.com/material-components/material-components-android/assets/78220358/e302e232-0dbb-4c0d-b808-ddbfb70e1092">

use **Theme.testApllication** and wrong **ThemeOverlay.App.Button**:
<img width="263" alt="image" src="https://github.com/material-components/material-components-android/assets/78220358/f21cd7e3-de90-4594-838c-e672858edbe7">

use **Theme.testApllication** and right **ThemeOverlay.App.Button**:
<img width="264" alt="image" src="https://github.com/material-components/material-components-android/assets/78220358/5d656e69-2e3c-4fca-8f38-29e686955ad5">

Complete themes. xml:
```:XML
<resources xmlns:tools="http://schemas.android.com/tools">
    <!-- Base application theme. -->
    <style name="Theme.TextApplication" parent="Theme.Material3.Light.NoActionBar">
        <!-- Primary brand color. -->
        <item name="colorPrimary">@color/purple_500</item>
        <item name="colorPrimaryVariant">@color/purple_700</item>
        <item name="colorOnPrimary">@color/white</item>
        <!-- Secondary brand color. -->
        <item name="colorSecondary">@color/teal_200</item>
        <item name="colorSecondaryVariant">@color/teal_700</item>
        <item name="colorOnSecondary">@color/black</item>
        <!-- Status bar color. -->
        <item name="android:statusBarColor">?attr/colorPrimaryVariant</item>
        <!-- Customize your theme here. -->
        <item name="toolbarStyle">@style/Widget.App.Toolbar</item>
        <item name="borderlessButtonStyle">@style/Widget.App.Button.TextButton</item>
        <item name="materialButtonOutlinedStyle">@style/Widget.App.Button.OutlinedButton</item>
        <item name="materialButtonStyle">@style/Widget.App.Button</item>

    </style>

    <style name="Widget.App.Toolbar" parent="Widget.Material3.Toolbar">
        <item name="materialThemeOverlay">@style/ThemeOverlay.App.Toolbar</item>
        <item name="titleTextAppearance">@style/TextAppearance.Material3.TitleLarge</item>
        <item name="subtitleTextAppearance">@style/TextAppearance.Material3.TitleMedium</item>
    </style>

    <style name="ThemeOverlay.App.Toolbar" parent="">
        <item name="titleTextColor">@color/teal_700</item>
        <item name="colorSurface">#FF660088</item>
        <item name="colorOnSurface">#FF667788</item>
        <item name="android:background">@color/black</item>
    </style>

    <style name="Widget.App.Button.TextButton" parent="Widget.Material3.Button.TextButton">
        <item name="materialThemeOverlay">@style/ThemeOverlay.App.Button.TextButton</item>
    </style>

    <style name="Widget.App.Button.OutlinedButton" parent="Widget.Material3.Button.OutlinedButton">
        <item name="materialThemeOverlay">@style/ThemeOverlay.App.Button.TextButton</item>
    </style>

    <style name="Widget.App.Button" parent="Widget.Material3.Button">
        <item name="materialThemeOverlay">@style/ThemeOverlay.App.Button</item>
    </style>

    <style name="ThemeOverlay.App.Button.TextButton" parent="">
        <item name="colorPrimary">@color/teal_700</item>
    </style>

    <style name="ThemeOverlay.App.Button" parent="">
<!--        <item name="colorPrimary">@color/teal_700</item>-->
<!--        <item name="colorOnPrimary">@color/black</item>-->
        <item name="colorContainer">@color/teal_700</item>
        <item name="colorOnContainer">@color/black</item>
    </style>
</resources>
```

Android API version: 33

Material Library version: 1.9.0

Device: Redmi Note 11 5G

**Comment by imhappi** [comment]
Hi @jimie-chaos can you explain what this bug is about in English?

**Comment by jimie-chaos** [comment]
> 嗨，你能用英语解释一下这个错误是什么吗？

I edit it,pls read again

**Comment by imhappi** [comment]
Hi @jimie-chaos ! Our docs are incorrect, we'll update them. Thanks for bringing it to our attention

The MaterialButton uses `colorContainer` and `colorOnContainer` to control the colors and those attributes should be used.

Normally `colorPrimary` is the default value of `colorContainer`, but this is actually set with `materialThemeOverlay` in `Widget.Material3.Button`, so overriding `materialThemeOverlay` in your own style actually removes this relationship with colorContainer and colorPrimary

tl;dr please use `colorContainer` and `colorOnContainer` if you want to override MaterialButton colors using theme overlays :)
